gef.py

```diff
@@ -3257,6 +3304,213 @@ def del_setting(self, name):
 #     def do_invoke(self, argv):
 #         return
 
+@register_command
+class PieCommand(GenericCommand):
+    """PIE breakpoint support"""
+
+    _cmdline_ = "pie"
+    _syntax_  = "{:s} (breakpoint|info|delete|run|attach|remote)".format(_cmdline_)
+
+    def __init__(self):
+        super(PieCommand, self).__init__(prefix=True)
+        return
+
+    def do_invoke(self, argv):
+        if len(argv)==0:
+            self.usage()
+            return
+
+
+@register_command
+class PieBreakpointCommand(GenericCommand):
+    """Set a PIE breakpoint"""
+
+    _cmdline_ = "pie breakpoint"
+    _syntax_  = "{:s} BREAKPOINT".format(_cmdline_)
+
+    def do_invoke(self, argv):
+        global __pie_counter__, __pie_breakpoints__
+        if (len(argv) < 1):
+            self.usage()
+            return
+        bp_expr = ' '.join(argv)
+        if bp_expr.startswith("*"):
+            # Handling pointer
+            tmp_bp_expr = bp_expr
+            bp_expr = bp_expr[1:].replace(' ', '')
+            if bp_expr.startswith('0x') and all([c in string.hexdigits for c in bp_expr[2:]]):
+                # hex string
+                addr = int(bp_expr[2:], 16)
+                self.set_pie_breakpoint(lambda base: "b *{}".format(base + addr), addr)
+            elif bp_expr.isdigit():
+                # number
+                addr = int(bp_expr[2:])
+                self.set_pie_breakpoint(lambda base: "b *{}".format(base + addr), addr)
+            else:
+                # all else should be symbols
+                bp_expr = tmp_bp_expr
+                self.set_pie_breakpoint(lambda base: "b {}".format(bp_expr), bp_expr)
+        else:
+            # symbols
+            self.set_pie_breakpoint(lambda base: "b {}".format(bp_expr), bp_expr)
+
+    @staticmethod
+    def set_pie_breakpoint(set_func, addr):
+        global __pie_counter__, __pie_breakpoints__
+        __pie_breakpoints__[__pie_counter__] = PieVirtualBreakpoint(set_func, __pie_counter__, addr)
+        __pie_counter__ += 1
+                
+
+@register_command
+class PieInfoCommand(GenericCommand):
+    """Display breakpoint info"""
+
+    _cmdline_ = "pie info"
+    _syntax_  = "{:s} BREAKPOINT".format(_cmdline_)
+
+    def do_invoke(self, argv):
+        global __pie_breakpoints__
+        if len(argv) < 1:
+            # No breakpoint info needed
+            bps = [__pie_breakpoints__[x] for x in __pie_breakpoints__]
+        else:
+            try:
+                bps = map(
+                    lambda x: __pie_breakpoints__[int(x)],
```

Comment:
You keep using this same lambda a lot, consider putting it as a `staticmethod` instead.